### PR TITLE
refactor: Rename CourseEditLTIFieldsEnabledFlag to CourseAllowPIISharingInLTIFlag and use it for LTI1.3 [BD-38] [BB-3899] [TNL-8104]

### PR DESCRIPTION
**Pull request by xitij2000**
This commit renames the CourseEditLTIFieldsEnabledFlag to CourseAllowPIISharingInLTIFlag since the aim is to expand its scope to all LTI-related PII sharing. It also removes the current LTI1.3 waffle flag for PII sharing.

[TNL-8104](https://openedx.atlassian.net/browse/TNL-8104)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @xitij2000! I've created [OSPR-5864](https://openedx.atlassian.net/browse/OSPR-5864) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: README.rst [hunk 1]
```diff
@@ -318,6 +318,15 @@ Please do not report security issues in public. Send security concerns via email
 Changelog
 =========
 
+2.12.0 - 2021-06-16
+-------------------
+
+* Rename `CourseEditLTIFieldsEnabledFlag` to `CourseAllowPIISharingInLTIFlag`
+  highlight its increased scope.
```
Comment (nedbat) [inline README.rst:325]:
```suggestion
  to highlight its increased scope.
```

File: README.rst [hunk 2]
```diff
@@ -318,6 +318,15 @@ Please do not report security issues in public. Send security concerns via email
...
+* Use `CourseAllowPIISharingInLTIFlag` for LTI1.3 in lieu of a the current
```
Comment (nedbat) [inline README.rst:326]:
```suggestion
* Use `CourseAllowPIISharingInLTIFlag` for LTI1.3 in lieu of the current
```

File: lti_consumer/models.py [hunk 3]
```diff
@@ -535,10 +536,14 @@ class Meta:
...
+class CourseAllowPIISharingInLTIFlag(ConfigurationModel):
     """
-    Enables the editing of "request username" and "request email" fields
-    of LTI consumer for a specific course.
+    Enables the sharing of PII via LTI for the specific course.
+
+    Depending on where it's used, further action might be needed to actually
+    enable sharing PII. For instance, in the LTI XBlock setting this flag
+    will allow editing the "request username" and "request email" fields, which
+    will also ned to be set to actually share PII.
```
Comment (nedbat) [inline lti_consumer/models.py:546]:
```suggestion
    will also need to be set to actually share PII.
```

**Review by nedbat — CHANGES_REQUESTED** [review]
A few typos.

**Comment by nedbat** [comment]
@giovannicimolin any concerns?

**Comment by asadazam93** [comment]
@xitij2000 Can you please create a TNL ticket for this and add a detailed description?

**Comment by asadazam93** [comment]
@xitij2000 How do you plan to backfill data?

**Comment by xitij2000** [comment]
> @xitij2000 How do you plan to backfill data?

I'm not sure I follow. What data needs to be backfilled?

**Comment by asadazam93** [comment]
> > @xitij2000 How do you plan to backfill data?
> 
> I'm not sure I follow. What data needs to be backfilled?

You are changing the name of the model. How do you plan to fill data from the previous model into the new model?

**Comment by xitij2000** [comment]
@asadazam93 Changing the name of a model doesn't do anything to the data. The table is renamed, so the data is retained. In ths case, since the table name is also set, it won't be renamed either.

**Review by awaisdar001 — APPROVED** [review]
This looks good to me 👍

**Review by asadazam93 — APPROVED** [review]

**Comment by asadazam93** [comment]
@xitij2000 This PR is still missing a TNL ticket. Can you please create one and add to the description? Let us merge the PR after the ticket is created

**Comment by xitij2000** [comment]
@asadazam93  I've added the ticket number to the PR title.

**Comment by awaisdar001** [comment]
Thanks, @xitij2000 can this be merged?

**Comment by xitij2000** [comment]
@awaisdar001 Yes, however https://github.com/edx/edx-platform/pull/26982 will need to be adjusted right after, so I would prefer if we do this during early IST hours.

**Comment by asadazam93** [comment]
@xitij2000 Let us merge both PRs during our meeting tomorrow morning?

**Comment by xitij2000** [comment]
> @xitij2000 Let us merge both PRs during our meeting tomorrow morning?

That might not be possible. After this PR is merged, it would still be nice to run the edx-platform tests to make sure nothing is broken. That would take some time.

**Comment by awaisdar001** [comment]
> > @xitij2000 Let us merge both PRs during our meeting tomorrow morning?
> 
> That might not be possible. After this PR is merged, it would still be nice to run the edx-platform tests to make sure nothing is broken. That would take some time.

we can merge this PR in standup -- and secondly we should get in touch on Slack when the other PR in edx-platform needs to be merged.

**Pull request merged**